### PR TITLE
Update Source file to pass Tests

### PR DESCRIPTION
**Pull request by abdulsemiu-atanda**
### What does this PR do

Update source file such that all Test Suite passes

### How to manually test this

Check the badges

### CODE REVIEW

**Comment by femioladeji** [comment]
LGTM

File: src/stringClass.js [hunk 1]
```diff
@@ -1,33 +1,165 @@
...
 
-  isQuestion() { },
+/**
+   * This method takes a string and checks if it's a question
+   * @param {String} this
+   * @return {Boolean} true or false
+   */
+  isQuestion() {
+    return /\?$/.test(this);
+  },
```
Comment (oredavids) [inline src/stringClass.js:49]: - You may want to take care of  more edge cases. `'?'.isQuestion()` , `'???'.isQuestion()` or `'   ?'.isQuestion()` will all return `true`  based on the present test.

File: src/stringClass.js [hunk 2]
```diff
@@ -1,33 +1,165 @@
...
 
-  words() { },
+/**
+   * This method takes a string and check if it has multiple words
+   * @param {String} this
+   * @return {Array} array
+   */
+  words() {
+    return this.replace(/[^\w\s]/g, '').split(/\s+/);
+  },
```
Comment (oredavids) [inline src/stringClass.js:58]: - You may also want to test more edge cases here as well.  What will `"%$$33 ^^^ *** Nigeria need*s a mira%%%%cle!".words()` return?

File: src/stringClass.js [hunk 3]
```diff
@@ -1,33 +1,165 @@
...
 
-  toCurrency() { },
+/**
+   * This method takes a string in currency format and converts to float type
+   * @param {String} this
+   * @return {String} this
+   */
+  fromCurrency() {
+    return this.replace(/,/g, '');
+  },
```
Comment (oredavids) [inline src/stringClass.js:91]: - You may want to take care of insignificant zeroes here. e.g `'123,456.00'.fromCurrency()` should return `'123456'`

File: src/stringClass.js [hunk 4]
```diff
@@ -1,33 +1,165 @@
...
 
-  isDigit() { },
+/**
+   * This method takes a string and checks if it's a single digit
+   * @param {String} this
+   * @return {Boolean} true or false
+   */
+  isDigit() {
+    return /^\d$/g.test(this);
+  },
```
Comment (oredavids) [inline src/stringClass.js:153]: - Also more edge cases. Will your test admit `-5` or `+2` ?

File: src/stringClass.js [hunk 5]
```diff
@@ -1,33 +1,165 @@
...
 
-  ucFirst() { },
+/**
+   * This method takes a string and returns its first letter in uppercase
+   * @param {String} this
+   * @return {String} this
+   */
+  ucFirst() {
+    return this.replace(this[0], this[0].toUpper());
```
Comment (thuo) [inline src/stringClass.js]: This will throw an error for  `ucFirst()` is called on an empty string because then `this[0]` will be `undefined` and calling a method on `undefined` is an error.

File: test/stringClassSpec.js [hunk 6]
```diff
@@ -48,6 +46,10 @@ describe('isQuestion', () => {
   it('should return false if string is not a question', () => {
     expect('rhy$th#?m'.isQuestion()).toBeFalsy();
   });
+
+  it('should return false if string contains no word character', () => {
+    expect('???'.isQuestion()).toBeFalsy();
+  });
 });
 
 describe('Words', () => {
```
Comment (thuo) [inline test/stringClassSpec.js:55]: A test that compares the return value of `words()` with a hard-coded array of the expected words would be useful in ensuring the method actually works. e.g `expect('two words'.words()).toEqual(['two', 'words'])`. You'd have several such tests for different edge cases.

File: test/stringClassSpec.js [hunk 7]
```diff
@@ -74,11 +76,19 @@ describe('To Currency', () => {
...
 describe('From Currency', () => {
   it('should return string in number format', () => {
-    expect('20,003.67'.toCurrency()).toEqual('20003.67');
+    expect('20,003.67'.fromCurrency()).toEqual('20003.67');
+  });
+
+  it('should return a number format without insignificant zero', () => {
+    expect('20,003.00'.fromCurrency()).toEqual('20003');
   });
 });
```
Comment (thuo) [inline test/stringClassSpec.js:99]: For `fromCurrency` and `toCurrency`, it would be useful to test with strings that are well formed currencies e.g having letters or misplaced commas and ensure that the implementation handles those edge cases.

File: test/stringClassSpec.js (hunk 6 above)
Comment (abdulsemiu-atanda) [reply test/stringClassSpec.js:55]: Updated the Test Suite for this. Thanks for pointing this out
